tensorflow_examples/models/mask_rcnn/model.py

```diff
@@ -0,0 +1,2928 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Masked R-CNN compatible with TF 2.0.
+
+Reference:
+[Mask R-CNN](https://arxiv.org/asb/1703.06870)
+
+Adapted from:
+[Matterport - Mask_RCNN](https://github.com/matterport/Mask_RCNN)
+
+"""
+
+import os
+import random
+import datetime
+import re
+import math
+import logging
+from collections import OrderedDict
+import multiprocessing
+import numpy as np
+import tensorflow as tf
+import tensorflow.keras.backend as KB
+import tensorflow.keras.layers as KL
+
+from mrcnn import utils
+
+# Requires TensorFlow 2.0
+assert tf.__version__.startswith('2')
+
+
+############################################################
+#  Utility Functions
+############################################################
+
+def log(text, array=None):
```

Comment:
This file is huge. Can you break this up?

For example, putting utility functions in utils.py, model in models.py, training in train.py, etc

Also, is it possible to add MirroredStrategy (https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/distribute/MirroredStrategy) to this model?